quinn/src/runtime.rs

```diff
@@ -31,10 +31,15 @@ pub trait AsyncTimer: Send + Debug + 'static {
 
 /// Abstract implementation of a UDP socket for runtime independence
 pub trait AsyncUdpSocket: Send + Sync + Debug + 'static {
-    /// Send UDP datagrams from `transmits`, or register to be woken if sending may succeed in the
-    /// future
-    fn poll_send(&self, cx: &mut Context, transmits: &[Transmit])
-        -> Poll<Result<usize, io::Error>>;
+    /// Create a helper for awaiting I/O readiness
+    fn create_io_poller(self: Arc<Self>) -> Pin<Box<dyn UdpPoller>>;
+
+    /// Send UDP datagrams from `transmits`, or return `WouldBlock` and clear the underlying
+    /// socket's readiness, or return an I/O error
+    ///
+    /// If this returns [`io::ErrorKind::WouldBlock`], [`UdpPoller::poll_writable`] must be called
+    /// to register the calling task to be woken when a send should be attempted again.
+    fn try_send(&self, transmit: &Transmit) -> io::Result<()>;
```

Comment:
I wish this would take `Transmit` owned (and if needed, returns `Transmit` in the error case)

Reply:
Why?

Reply:
because we need to rewrite these before sending

Reply:
As of this PR, `Transmit` no longer owns any non-`Copy` data. Just `Clone` it.

Reply:
fair enough 👍